### ISSUE TITLE
fix: predict_proba and predict are stochastic when the model is left in train mode

### ISSUE THREAD

**Issue by g8rdier** [g8rdier/hqnn-forge#59]
## Context

`predict_proba` and `predict` on both classifiers are decorated with `@torch.no_grad()`, and
`predict_proba` is documented as "inference mode". Neither method actually switches the module
to eval mode: `@torch.no_grad()` only disables autograd, and `nn.Dropout` still checks
`self.training`.

With `dropout_p > 0` and the model in train mode (its state after construction, and its normal
state inside a training loop), every call zeroes a random subset of the pre-head activations and
rescales the rest by `1 / (1 - p)`. Repeated calls on the same input return different
probabilities and labels. Nothing raises.

- `HybridBinaryClassifier` applies dropout to the quantum-layer output (`hybrid_classifier.py:226`).
- `ParallelHybridClassifier` applies dropout to the fused classical + quantum vector
  (`parallel_hybrid_classifier.py:276`).

## Steps to Reproduce

```python
import torch
from hqnn_forge.models import HybridBinaryClassifier, ParallelHybridClassifier

kw = dict(n_input_features=6, n_qubits=4, n_layers=1, dropout_p=0.5,
          device_name="default.qubit", diff_method="parameter-shift")
x = torch.randn(16, 6, generator=torch.Generator().manual_seed(0))

for cls in (HybridBinaryClassifier, ParallelHybridClassifier):
    torch.manual_seed(0)
    m = cls(**kw)                      # freshly constructed -> m.training is True
    runs = torch.stack([m.predict_proba(x) for _ in range(20)])
    print(cls.__name__, (runs.max(0).values - runs.min(0).values).max())
```

Measured over 20 calls on the same 16-sample batch:

| Model | Mode | Max prob spread | Mean prob spread | Samples whose label flips |
|---|---|---|---|---|
| `HybridBinaryClassifier` | train | 0.430 | 0.287 | 15 / 16 |
| `ParallelHybridClassifier` | train | 0.804 | 0.526 | 15 / 16 |
| either | eval | 0 | 0 | 0 / 16 |

## Affected Code

- `hqnn_forge/models/hybrid_classifier.py:233` (`predict_proba`) and `:252` (`predict`). This
  code predates #56.
- `hqnn_forge/models/parallel_hybrid_classifier.py:283` (`predict_proba`) and `:302` (`predict`).
  #56 copied the same pattern from the serial model.

No current caller hits this. `examples/quick_start.py` calls `model.eval()` before predicting, and
both class docstring examples use the default `dropout_p=0.0`, where dropout is `nn.Identity`.
It will affect anyone who evaluates mid-training without calling `eval()` first, for example the
per-epoch validation that early stopping in #49 needs.

## Expected Outcome

- `predict_proba` / `predict` return deterministic results whatever mode the caller left the model
  in. For example, switch to eval for the duration of the call and restore the previous
  `training` flag in a `try` / `finally`, so that predicting mid-training does not leave the model
  stuck in eval mode.
- Both classifiers are fixed together, so the shared `predict_proba` / `predict` API behaves the
  same on both.
- A test for each model: with `dropout_p > 0` and the model in train mode, two `predict_proba`
  calls on the same input are identical, and `model.training` is still `True` afterwards.

## Type

Bug: a public inference method gives non-deterministic predictions with no error. The serial
classifier already had it; found during review of #56.
